src/utils/internalUtils.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Utility function that are only meant to be used internally. It would not guarantee backward compatibility if they were used outside of the package.
+ */
+
+import AxiosErrorCodes from "../core/AxiosErrorCodes";
+
+export const isCoreServicesOrgUrlDNSError = (error: any, coreServicesOrgUrl: string | null, dynamicsLocationCode: string | null): boolean => { // eslint-disable-line @typescript-eslint/no-explicit-any
+    // Validating whether it's an 'ERR_NAME_NOT_RESOLVED' error
+    // `ERR_NETWORK` could return false positives since the error can be caused by network disconnection
+    const isDNSUrlResolveError = error.isAxiosError && error.code == AxiosErrorCodes.ERR_NETWORK;

```

Comment:
does this cover 500, 502, 404....?

Reply:
We’re only looking for “ERR_NAME_NOT_RESOLVED”. It only happens when the ip address cannot be resolved by the DNS which is irrespective of the response status code. Axios only returns “ERR_NETWORK” when this scenario happens